fifo-sqs-redrive/template.yaml

```diff
@@ -0,0 +1,145 @@
+AWSTemplateFormatVersion: '2010-09-09'
+Transform: AWS::Serverless-2016-10-31
+Description: SQS with DLQ and Lambda event source 
+
+Resources:
+  ## Define the SQS Queues ##
+  MyOriginalQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: MyOriginalQueue.fifo
+      FifoQueue: true
+      ContentBasedDeduplication: true
+      VisibilityTimeout: 30
+      MessageRetentionPeriod: 1209600
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt MyDeadLetterQueue.Arn
+        maxReceiveCount: 1
+
+  MyDeadLetterQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: MyDeadLetterQueue.fifo
+      FifoQueue: true
+      ContentBasedDeduplication: true
+      MessageRetentionPeriod: 1209600
+      RedriveAllowPolicy:
+        redrivePermission: "allowAll"
+
+  MyReProcessQueue:
+    Type: AWS::SQS::Queue
+    Properties:
+      QueueName: MyReProcessQueue.fifo
+      FifoQueue: true
+      ContentBasedDeduplication: true
+      VisibilityTimeout: 30
+      MessageRetentionPeriod: 1209600
+      RedrivePolicy:
+        deadLetterTargetArn: !GetAtt MyDeadLetterQueue.Arn
+        maxReceiveCount: 1
+
+
+  ## Define the Lambda Functions ##
+  MyOriginalQueuePollerFunction:
+    Type: AWS::Serverless::Function
+    Properties:
+      CodeUri: originalmessage/
+      Handler: app.lambda_handler
+      Runtime: python3.12
+      FunctionName: MyOriginalQueuePollerFunction
+      Policies:
+        - SQSPollerPolicy:
+            QueueName: !GetAtt MyOriginalQueue.QueueName
+      Environment:
+        Variables:
+          QUEUE_URL: !Ref MyOriginalQueue              
+      Events:
+        SqsEvent:
+          Type: SQS 
+          Properties:
+            Queue: !GetAtt MyOriginalQueue.Arn
+            BatchSize: 1
```

Comment:
Does the BatchSize need to be 1 in this case?

Reply:
Need not be 1. I have updated it to 10 now.